File: src/main/java/io/thisdk/github/ordering/service/UserService.java
package io.thisdk.github.ordering.service;

import io.thisdk.github.ordering.bean.CmsUser;
import io.thisdk.github.ordering.bean.LoginUser;
import io.thisdk.github.ordering.dao.impl.UserAccountDao;
import io.thisdk.github.ordering.exception.OrderingErrorInfoEnum;
import io.thisdk.github.ordering.exception.OrderingErrorInfoException;
import io.thisdk.github.ordering.utils.StringUtils;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class UserService {

    final
    TokenService tokenService;

    final
    UserAccountDao userAccountDao;

    @Autowired
    public UserService(TokenService tokenService, UserAccountDao userAccountDao) {
        this.tokenService = tokenService;
        this.userAccountDao = userAccountDao;
    }

    public String login(String username, String password) {
        CmsUser cmsUser = userAccountDao.getUserInfo(username, password);
        if (cmsUser != null) {
            LoginUser loginUser = new LoginUser(cmsUser);
            return tokenService.createToken(loginUser);
        } else throw new OrderingErrorInfoException(OrderingErrorInfoEnum.LOGIN_ERROR);

    }

    public CmsUser getCurrentUserInfo() {
        LoginUser loginUser = tokenService.getLoginUser(StringUtils.getRequest());
        return loginUser.getUser();
    }
}

File: src/main/kotlin/io/thisdk/github/ordering/service/UserService.kt
package io.thisdk.github.ordering.service

import io.thisdk.github.ordering.bean.User
import io.thisdk.github.ordering.dao.UserDao
import io.thisdk.github.ordering.exception.OrderingErrorInfoEnum
import io.thisdk.github.ordering.exception.OrderingErrorInfoException
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class UserService {

    @Autowired
    lateinit var userDao: UserDao

    fun queryByUserId(id: String): User {
        return userDao.queryByUserId(id) ?: throw OrderingErrorInfoException(OrderingErrorInfoEnum.USER_NOT_EXIST)
    }

    fun hasUserByUserName(username: String): Boolean {
        try {
            queryByUserName(username)
        } catch (exception: OrderingErrorInfoException) {
            exception.printStackTrace()
            return false
        }
        return true
    }

    fun queryByUserName(username: String): User {
        return userDao.queryByUsername(username)
            ?: throw OrderingErrorInfoException(OrderingErrorInfoEnum.USER_NOT_EXIST)
    }

    fun queryAllMiniProgramUser(): List<User> {
        return userDao.queryAllMiniProgramUser()
    }

    fun insertUser(user: User): User {
        return userDao.insert(user) ?: throw OrderingErrorInfoException(OrderingErrorInfoEnum.CREATE_USER_ERROR)
    }

    fun deleteByUserName(username: String): Boolean {
        return userDao.deleteByUsername(username)
    }

}